app/controllers/boats_controller.rb

```diff
@@ -14,6 +14,15 @@ def create
   end
 end
 
+def edit
+  @boat = current_user.boat
```

Comment:
current_user.boat doesn't exists (it would be something like current_user.boats)
on the other hand, you can use the params on the url to find the boat using... Boat.find(params[:id])